core/platform/src/raw_syscalls.rs

```diff
@@ -3,242 +3,51 @@
 
 /// `RawSyscalls` allows a fake Tock kernel to be injected into components for
 /// unit testing. It is implemented by `libtock_runtime::TockSyscalls` and
-/// `libtock_unittest::FakeSyscalls`. **Components should not use `RawSyscalls`
+/// `libtock_unittest::fake::Kernel`. **Components should not use `RawSyscalls`
 /// directly; instead, use the `Syscalls` trait, which provides higher-level
 /// interfaces to the system calls.**
+// -----------------------------------------------------------------------------
+// Full documentation for everything in this file is in doc/RawSyscallsDesign.md
+// at the root of this repository.
+// -----------------------------------------------------------------------------
```

Comment:
hmmm do you wanna do one of those fancy `#[doc]` annotations that pulls in an external md file?